activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-api/src/main/java/org/activiti/cloud/modeling/api/ModelValidator.java

```diff
@@ -15,42 +15,72 @@
  */
 package org.activiti.cloud.modeling.api;
 
+import java.util.Collection;
+
 /**
  * Business logic related with validation of a model
  */
 public interface ModelValidator extends ModelValidationErrorProducer {
 
-    /**
-     * Validate the given model.
-     * @param modelContent content of the model to validate
-     * @param validationContext the validation context
-     */
-    void validate(byte[] modelContent,
```

Comment:
Basically I think that what @LorenzoDiGiuseppe was asking here is that these `validate` methods (this one and the following) return a list of `ModelValidationError`

Then delegate in the implementations of the `ValidationStrategy` interface wether those errors are going to be thrown or not.

Doing this, we can avoid having to collect the exceptions like we do here:
https://github.com/Activiti/activiti-cloud/blob/b46c62242caa2accd9bd64e694d48621a29d6a15/activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/service/utils/AggregateErrorValidationStrategy.java#L35-L44

Is this what your enhancement was about @LorenzoDiGiuseppe ??

Reply:
Hi @pmartinezga, yes, this issue was for removing the needs to throw and to collect exceptions: https://github.com/Activiti/activiti-cloud/pull/965#discussion_r1088756734. 

In my opinion, we could probably remove the  `ValidationStrategy` since `ModelService` and `ProjectService` should be enough (there would be no more need to catch exceptions and aggregate them).

About the type of errors, I think just that we should not use `ModelValidationError` as super class since it is actually adopted for semantic errors only. If we decide to use it as base class for semantic and syntactic errors we should also change the implementations of `BpmnCommonModelValidator` because it would be like changing this interface signature.

I am including @erdemedeiros to the discussion, since he may have different ideas on this topic.

Reply:
Is `SyntacticModelError` used only in the case where we are not able to parse the json? If that's the case maybe we don't need it as per my comment [here](https://github.com/Activiti/activiti-cloud/pull/1017#discussion_r1124265548). And then, no need for specialisation, we only have `ModelValidationError` as before. WDYT?

Reply:
AFAIK, at the moment it is used just for mapping `JSONException` and when there is a problem converting `byte[]` to a model or model extension: [`ProcessExtensionsModelValidator`](https://github.com/Activiti/activiti-cloud/blob/9796c3a36c422613c42051df017bf7dc81e4330a/activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/extensions/ProcessExtensionsModelValidator.java#L124-L131), [`ExtensionsJsonSchemaValidator `](https://github.com/Activiti/activiti-cloud/blob/70202e7a5d612c1e44dffd2ec56df0b3c27e8cec/activiti-cloud-modeling-service/activiti-cloud-services-modeling/activiti-cloud-services-modeling-service/src/main/java/org/activiti/cloud/services/modeling/validation/extensions/ExtensionsJsonSchemaValidator.java#L65-L72). So it could still be good to keep throwing them and reduce the changes.